src/Components/Profile/UserCoverModal.js

```diff
@@ -0,0 +1,124 @@
+import { useState } from 'react';
+import './Modal.css'
+import { Formik, Form } from "formik";
+import {  AddToPhotos } from '@material-ui/icons';
+import { connect } from 'react-redux';
+import { UploadCoverPhoto } from '../../Auth/Actions/userActions';
+import { Audio } from  'react-loader-spinner'
+import { toast } from 'react-toastify'
+
+
+
+const UserCoverModal = ({UploadCoverPhoto, setCoverImg}) => {
+    const [preview, setPreview] = useState([])
+    const handlePhotoChange = (e, setFieldValue) =>{
+        const files = e.target.files[0];
+            setFieldValue("photo", files);
+            if(files) {
+            const reader = new FileReader()
+                reader.readAsDataURL(files)
+                reader.onload = () => {
+                    setPreview(reader.result)
+                }
+            }
+    }
+    
+    const handleClick = (e) => {
+        if (e.target.classList.contains("dismiss")) {
+        setCoverImg(null);
+        }
+    };
+
+    const handleCancle = (e)=>{
+        if (e.target){
+            setPreview([])
+        }
+    }
+
+
+  return (
+    <div className="overlay dismiss" onClick={handleClick}>
+        <div className='modal_container'>
+            <div className="upload_profile_right">
+                <div className='modal_top'>
+                    <h3>Upload Cover Photo</h3>
+                    <span className="dismiss button cover " onClick={handleClick}>X</span>
+                </div>
+                <hr className='upload_hr'/>
+            </div>
+            <Formik
+                initialValues={{
+                    photo: ''
+                }}
+                onSubmit={(values, {setSubmitting,}) => {
+                    let formData = new FormData();
+                    formData.append(`photo`, values.photo);
+                    UploadCoverPhoto(formData, setSubmitting).then(response => {
```

Comment:
now think about this, you don't need to pass setSubmitting you can do it inside the .then